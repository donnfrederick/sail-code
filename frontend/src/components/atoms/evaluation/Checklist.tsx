import * as React from 'react'
import styled from 'styled-components'

import resolvePath from 'utils/resolvePath'

interface Props {
  text: string
}
export default ({ text }: Props) => {
  const checkSrc = resolvePath.image('common/evaluation/check_red@2x.png')
  return (
    <Container>
      <Check src={checkSrc} />
      <Description>{text}</Description>
    </Container>
  )
}

interface CenterProps {
  isCenter?: boolean
}

const Container = styled<CenterProps, any>('div')`
  height: 41px;
  width: 100%;
  margin-bottom: 17px;
  display: flex;
  ${({ isCenter }) => isCenter && 'justify-content: center'};
  align-items: center;
  background: none;
  color: rgb(65, 87, 101);
`

const Check = styled.img`
  margin-right: 10px;
  width: 23px;
`

const Description = styled.span`
  font-size: 28px;
`
