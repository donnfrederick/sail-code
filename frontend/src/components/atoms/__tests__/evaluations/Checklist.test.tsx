import Checklist from 'components/atoms/evaluation/Checklist'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

test('render <Checklist />', () => {
  const tree = renderer
    .create(<Checklist text="このユーザーをブロックする" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
