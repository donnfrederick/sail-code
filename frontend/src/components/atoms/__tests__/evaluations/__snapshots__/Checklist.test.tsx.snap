// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Checklist /> 1`] = `
<div
  className="sc-bdVaJa jQyLZh"
>
  <img
    className="sc-bwzfXH chcSXJ"
    src="/assets/img/common/evaluation/check_red@2x.png"
  />
  <span
    className="sc-htpNat vsbMt"
  >
    このユーザーをブロックする
  </span>
</div>
`;
